server/src/com/cloud/network/router/VpcVirtualNetworkApplianceManagerImpl.java

```diff
@@ -455,16 +471,26 @@ protected void finalizeNetworkRulesForNetwork(final Commands cmds, final DomainR
 
         super.finalizeNetworkRulesForNetwork(cmds, domainRouterVO, provider, guestNetworkId);
 
+        final VpcVO vpc = _vpcDao.findById(domainRouterVO.getVpcId());
         if (domainRouterVO.getVpcId() != null) {
-
             if (domainRouterVO.getState() == State.Starting || domainRouterVO.getState() == State.Running) {
                 if (_networkModel.isProviderSupportServiceInNetwork(guestNetworkId, Service.NetworkACL, Provider.VPCVirtualRouter)) {
                     final List<NetworkACLItemVO> networkACLs = _networkACLMgr.listNetworkACLItems(guestNetworkId);
                     if (networkACLs != null && !networkACLs.isEmpty()) {
-                        s_logger.debug("Found " + networkACLs.size() + " network ACLs to apply as a part of VPC VR " + domainRouterVO + " start for guest network id=" + guestNetworkId);
+                        s_logger.debug(
+                                "Found " + networkACLs.size() + " network ACLs to apply as a part of VPC VR " + domainRouterVO + " start for guest network id=" + guestNetworkId);
                         _commandSetupHelper.createNetworkACLsCommands(networkACLs, domainRouterVO, cmds, guestNetworkId, false);
                     }
                 }
+                if (_vpcOffServiceDao.areServicesSupportedByNetworkOffering(vpc.getVpcOfferingId(), Service.VPCDynamicRouting)
+                        && _networkModel.areServicesSupportedInNetwork(guestNetworkId, Service.VPCDynamicRouting)) {
+                    try {
+                        _commandSetupHelper.createQuaggaConfigCommand(domainRouterVO, vpc.getId(), cmds);
+                    } catch (BadCIDRException ex) {
+                        s_logger.debug(ex);
+                        throw new CloudRuntimeException("The cidr for dynamic routing is bad " + ex);
```

Comment:
Please add the ``ex`` as the cause to provide a complete stack trace for debugging.